config.go

```diff
@@ -46,4 +46,7 @@ type Config struct {
 
 	// Interfaces will override the interfaces used for queries and answers.
 	Interfaces []net.Interface
+
+	// Override the default behavior of echoing the query with the answer
+	DoNotEchoQueryWithAnswer bool
```

Comment:
Having this in the Config looks ugly to me, however, I couldn't see how to be able to unit test both modes without doing this or a very radical restructuring.